ocs_ci/deployment/vmware.py

```diff
@@ -24,23 +32,302 @@ def __init__(self):
         """
         super(VSPHEREBASE, self).__init__()
         self.region = config.ENV_DATA.get('region')
+        self.server = config.ENV_DATA.get('vsphere_server')
+        self.user = config.ENV_DATA.get('vsphere_user')
+        self.password = config.ENV_DATA.get('vsphere_password')
+        self.cluster = config.ENV_DATA.get('vsphere_cluster')
+        self.datacenter = config.ENV_DATA.get('vsphere_datacenter')
+        self.datastore = config.ENV_DATA.get('vsphere_datastore')
+        self.vsphere = VSPHEREUtil(self.server, self.user, self.password)
+
+    def attach_disk(self, size=100):
+        """
+        Add a new disk to all the workers nodes
+
+        Args:
+            size (int): Size of disk in GB (default: 100)
+
+        """
+        vms = self.vsphere.get_all_vms_in_pool(
+            config.ENV_DATA.get("cluster_name"),
+            self.datacenter,
+            self.cluster
+        )
+        # Add disks to all worker nodes
+        for vm in vms:
+            if "compute" in vm.name:
+                self.vsphere.add_disks(
+                    config.ENV_DATA.get("extra_disks", 1),
+                    vm,
+                    size,
+                    constants.VM_DISK_TYPE
+                )
 
 
 class VSPHEREUPI(VSPHEREBASE):
     """
     A class to handle vSphere UPI specific deployment
     """
     def __init__(self):
-        self.name = self.__class__.__name__
         super(VSPHEREUPI, self).__init__()
+        self.ipam = config.ENV_DATA.get('ipam')
```

Comment:
if all of those are required I wouldn't use the .get as it's return None if it's not defined but I guess those are required variables. So would access those keys directly with config.ENV_DATA['imap'] and for the rest required. So if it's not defined you will fail with key error and user will know that this parameter is required for vmware deployment.  If it won't be set and it will be None you will continue with some other action and will fail later on and will be hard to debug the issue.